app/src/main/java/com/example/levi9application/CocktailsFragment.kt

```diff
@@ -31,12 +40,45 @@ class CocktailsFragment : Fragment(R.layout.fragment_cocktails) {
         cocktailViewModel = ViewModelProvider(this)[CocktailViewModel::class.java]
 
 
+        requireActivity().addMenuProvider(object : MenuProvider {
+            override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
+                menuInflater.inflate(R.menu.toolbar_menu, menu)
+            }
+
+            override fun onMenuItemSelected(menuItem: MenuItem): Boolean {
+                when (menuItem.itemId) {
+                    R.id.menuSearch -> {
+                        if (binding.etSearch.visibility != View.VISIBLE) binding.etSearch.visibility =
+                            View.VISIBLE
+                        else binding.etSearch.visibility = View.GONE
+                    }
+
+                    R.id.menuFilter -> Toast.makeText(
+                        activity, "Clicked on Filter", Toast.LENGTH_SHORT
+                    ).show()
+                }
+                return true
+            }
+
+        }, viewLifecycleOwner, Lifecycle.State.RESUMED)
+
+        binding.etSearch.doAfterTextChanged {
+            query = it.toString().trim()
+            cocktailViewModel.getCocktails(query)
+        }
+
+        binding.swipeRefreshLayout.setOnRefreshListener {
```

Comment:
Please change logic so that if we have some text typed in the search bar and we pull to refresh, to return the data based on the search text